internal/core/workers/worker.go

```diff
@@ -44,10 +45,13 @@ type Worker interface {
 // its construction. This struct is going to be used to inject the worker
 // dependencies like ports.
 type WorkerOptions struct {
-	OperationManager   ports.OperationManager
-	OperationExecutors map[string]operations.Executor
-	RoomManager        ports.RoomManager
-	Runtime            ports.Runtime
+	OperationManager              ports.OperationManager
+	OperationExecutors            map[string]operations.Executor
+	RoomManager                   ports.RoomManager
+	Runtime                       ports.Runtime
+	RoomStorage                   ports.RoomStorage
+	InstanceStorage               ports.GameRoomInstanceStorage
+	MetricsReporterIntervalMillis time.Duration
```

Comment:
Why did we decide to use this `MetricsReporterIntervalMillis` instead of using the `config` approach we are using on managers? It feels to me that if we want to add new configuration variables it'll be harder